### PR TITLE
Introduce goals collection to reduce complexity of Experiment#save

### PR DESCRIPTION
**Pull request by pakallis**

### CODE REVIEW

**Comment by andrew** [comment]
Thanks for the refactor @pakallis, would you be able to add some tests for the `goals_collection` class?

**Comment by pakallis** [comment]
Sure, i'm on it

File: split.gemspec [hunk 1]
```diff
@@ -30,4 +30,5 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'rack-test',   '~> 0.6'
   s.add_development_dependency 'rake',        '~> 11.1'
   s.add_development_dependency 'rspec',       '~> 3.4'
+  s.add_development_dependency 'pry'
```
Comment (pakallis) [inline split.gemspec]: I also added pry gem for easier debugging

Comment (andrew) [inline split.gemspec]: Can you add a broad version range here for pry?

**Pull request merged**